docs/website/docs/v1.x/dsl/socket/socket.md

```diff
@@ -0,0 +1,166 @@
+---
+title: "Socket"
+sidebar_label: "Socket"
+---
+
+`Socket[-R, +E, -A, +B]` models a function from `A` to `ZStream[R, E, B]`. When a value of type `A` is evaluated against a `Socket[R, E, A, B]`, it can either succeed with a `B`, or fail with an `E`, and it could have its requirement on `R`.
+
+## Creating Sockets
+
+### An empty Socket
+
+To create an empty Socket, you can use the `empty` constructor.
+
+```scala
+val socket = Socket.empty
+```
+
+### Socket that has ended
+
+To create a Socket that has ended, you can use the `end` constructor.
+
+```scala
+val socket = Socket.end
+```
+
+### Socket that always succeeds
+
+You can use the `succeed` constructor to create a Socket that always returns the same response and never fails.
+
+```scala
+val socket = Socket.succeed(WebSocketFrame.text("Hello, from ZIO-HTTP"))
+```
+
+### Socket that echoes the message
+
+To create a Socket that always echoes the message back, you can use the `echo` constructor.
+
+```scala
+val socket = Socket.echo(WebSocketFrame.text("Hello, from ZIO-HTTP"))
+```
+
+### Socket from a partial function
+
+`Socket.collect` can create a `Socket[R, E, A, B]` from a `PartialFunction[A,B]`.
+
+```scala
+val fromCollect = Socket.collect[WebSocketFrame] {
+  case WebSocketFrame.Text("fail") => ZStream.fail(new Exception("error"))
+  case WebSocketFrame.Text(text)   => ZStream.succeed(text)
+}
+```
+
+### Socket from a function
+
+To create a Socket from a function, you can use the `fromFunction` constructor.
+
+```scala
+val socket = Socket.fromFunction[WebSocketFrame](wsf => ZStream.succeed(wsf))
+```
+
+### Socket from a ZStream
+
+To create a socket from a `ZStream[R,E,B]`, you can use the `fromStream` constructor.
+
+```scala
+val transducer = ZTransducer[Int].map(elem => WebSocketFrame.Text(elem.toString))
+val stream     = ZStream
+  .fromIterable((0 to 10))
+  .transduce(transducer)
+
+val socket = Socket.fromStream(stream)
+```
+
+## Composing Sockets
+
+### Using `merge`
+
+You can merge to Sockets using the `merge` operator, the resulting Socket will emit the values of both Sockets.
+
+```scala
+val s1 = Socket.succeed(WebSocketFrame.text("Hello, from ZIO-HTTP"))
+val s2 = Socket.succeed(WebSocketFrame.text("Welcome to the party"))
+val socket = s1 merge s2
+```
+
+## Transforming Sockets
+
+### `map` over a Socket's output channel
+
+Socket is a monad, so you can use `map` to transform the output of a Socket from type `Socket[R,E,A,B]` to type `Socket[R,E,A,C]`, it takes a function from `B => Socket[R,E,A,C]`.
```

Comment:
```suggestion
Socket is a functor, so you can use `map` to transform the output of a Socket from type `Socket[R,E,A,B]` to type `Socket[R,E,A,C]`, it takes a function from `B => Socket[R,E,A,C]`.
```
Socket by definition looks more like a Functor. Need feedback.

Ref: https://medium.com/beingprofessional/understanding-functor-and-monad-with-a-bag-of-peanuts-8fa702b3f69e

Reply:
Use Socket is a `domain` for now to maintain consistency in the docs.